modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/freemarker/tool/java/parser/ResourceOpenAPIParser.java

```diff
@@ -618,6 +618,14 @@ private static List<JavaMethodParameter> _getJavaMethodParameters(
 				continue;
 			}
 
+			if ((StringUtil.equals(parameterName, "fields") ||
```

Comment:
I think in this case is enough to check just the first condition, like `StringUtil.equals(parameterName, "fields")` and you can add them as an or condition to the *if* in line 605.

In the case of page and pageSize is more complicated because you need to be sure that the two parameters exists at the same time

Reply:
+1